.github/workflows/run-tests.yml

```
@@ -135,8 +135,54 @@ jobs:
             -t "${COMMIT}" | tee -a "$GITHUB_ENV"
         shell: bash
 
+  pre-build:
+    name: Pre-build for cache
+    strategy:
+      fail-fast: false
+      matrix:
+        runs-on: [ubuntu-20.04]
+    runs-on: ${{ matrix.runs-on }}
```

nit: you can compress this into
```suggestion
    runs-on: ubuntu-20.04
```

---

oh, I copied that from misc-checks. not sure why it was like that. I'll fix them both

---

hmm, actually all the tests except functional-test were doing that. I simplified them all (except functional-test). even for that one, I'm not sure it needs to be a matrix value since it only ever has one value, I think it could just be a plain output, but I left it for now. @carlydf